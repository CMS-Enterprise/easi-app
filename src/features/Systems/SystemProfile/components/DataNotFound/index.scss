@use 'uswds-core' as *;
@use 'colors' as *;

.data-not-found {
  background-color: color('warning-lighter');
  border: 1px solid color('warning-light');
  border-radius: 0.25rem;
  border-radius: radius('md');
}