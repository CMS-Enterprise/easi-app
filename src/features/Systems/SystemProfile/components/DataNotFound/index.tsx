import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Icon, Link } from '@trussworks/react-uswds';

import './index.scss';

const DataNotFound = () => {
  const { t } = useTranslation('systemProfile');
  return (
    <div className="data-not-found padding-3 easi-body-medium">
      <div className="display-flex flex-justify">
        <h3 className="margin-top-1 margin-bottom-0">
          {t('dataNotFound.heading')}
        </h3>
        <Icon.Warning
          className="margin-left-1 text-warning top-1"
          aria-hidden
          size={3}
        />
      </div>
      <p>{t('dataNotFound.description')}</p>
      <Trans
        i18nKey="systemProfile:dataNotFound.contact"
        components={{
          emailLink: (
            <Link href="mailto:EnterpriseArchitecture@cms.hhs.gov">
              EnterpriseArchitecture@cms.hhs.gov
            </Link>
          )
        }}
      />
    </div>
  );
};

export default DataNotFound;
