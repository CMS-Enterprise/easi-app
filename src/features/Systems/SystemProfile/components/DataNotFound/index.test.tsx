import React from 'react';
import { render } from '@testing-library/react';

import DataNotFound from '.';

describe('DataNotFound component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<DataNotFound />);
    expect(asFragment()).toMatchSnapshot();
  });
});
