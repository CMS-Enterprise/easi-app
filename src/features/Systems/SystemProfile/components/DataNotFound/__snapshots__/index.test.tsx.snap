// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataNotFound component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="data-not-found padding-3 easi-body-medium"
  >
    <div
      class="display-flex flex-justify"
    >
      <h3
        class="margin-top-1 margin-bottom-0"
      >
        Data not currently available
      </h3>
      <svg
        aria-hidden="true"
        class="usa-icon usa-icon--size-3 margin-left-1 text-warning top-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
    </div>
    <p>
      EASi is currently having difficulty accessing some or all data for this section, so it cannot be displayed. Please check back later.
    </p>
    If you have any questions, please contact Enterprise Architecture at 
    <a
      class="usa-link"
      href="mailto:EnterpriseArchitecture@cms.hhs.gov"
    >
      EnterpriseArchitecture@cms.hhs.gov
    </a>
    .
  </div>
</DocumentFragment>
`;
